lib/CAPI/Dialect/FIRRTL.cpp

```diff
@@ -7,17 +7,21 @@
 //===----------------------------------------------------------------------===//
 
 #include "circt-c/Dialect/FIRRTL.h"
+#include "../../Dialect/FIRRTL/Import/FIRAnnotations.h"
```

Comment:
It's a little unusual to depend on a header in the lib directory. I'm not sure if it's worth refactoring this, or if it's fine to depend on the implementation in lib/Dialect/FIRRTL/Import from the CAPI.

Reply:
This is not fine, and headers in `lib` are there because they're not part of the library's interface.

Please move the declaration to an appropriate header location.

(although one might reasonably wonder if perhaps there's a reason this wasn't already there....)

Reply:
Added a simple test for the new function and moved `FIRAnnotations.h` file from `lib/Dialect/FIRRTL/Import/` to `include/circt/Dialect/FIRRTL/Import/`.

Thanks for the advice @dtzSiFive and the review from @mikeurbach :+1:

Reply:
Sorry, I don't think simply moving the header into the include directory is what @dtzSiFive meant. The original problem was you were depending on one C++ API that was an implementation detail. But to address that, you have made the entire set of C++ implementation details public APIs. Please don't do that either.

I think what you could do is revert the last commit, and think about in which header the declaration of fromJSONRaw should live, and in which library the definition should live. For example, one idea is to create a new `include/circt/Dialect/FIRRTL/Import/FIRAnnotations.h`, only define fromJSONRaw, and keep the definition in the CIRCTImportFIRFile library.

Reply:
My bad, I misunderstood the comment. In the latest commit, I only moved the function `fromJSONRaw` to the new header `include/circt/Dialect/FIRRTL/Import/FIRAnnotations.h`, and I renamed it to `importAnnotationsFromJSONRaw` in order to make it cleaner in the public namespace.

Hopefully I'm on the right path this time.

Reply:
Thanks, yes, I think that is more in line with what I was thinking. @dtzSiFive does it make sense to you now to expose this one function from the existing library?

Reply:
If we're exposing this functionality, this seems the way to do it.  Thanks! :+1: